server/base/src/main/java/org/apache/accumulo/server/compaction/FileCompactor.java

```diff
@@ -212,9 +216,21 @@ public CompactionStats call() throws IOException, CompactionCanceledException {
     try {
       FileOperations fileFactory = FileOperations.getInstance();
       FileSystem ns = this.fs.getFileSystemByPath(outputFile.getPath());
-      mfw = fileFactory.newWriterBuilder()
+
+      boolean dropCacheBehindMajcOutput = false;
+      if (!RootTable.ID.equals(this.extent.tableId())
+          && !MetadataTable.ID.equals(this.extent.tableId())
+          && acuTableConf.getBoolean(Property.TABLE_MAJC_OUTPUT_DROP_CACHE)) {
```

Comment:
@keith-turner - I tried looking through new pluggable compaction service code, I'm thinking it may be possible for someone to implement their own CompactionConfigurer such that they can set this new property based on the KeyExtent that is getting compacted. Is that possible? If not, then we may want to add that functionality in a follow-on commit. I'm thinking of the case where you have a date sorted table and you may not want to drop the files from the page cache when compacting recent data, but may want to for older data.

Reply:
> @keith-turner - I tried looking through new pluggable compaction service code, I'm thinking it may be possible for someone to implement their own CompactionConfigurer such that they can set this new property based on the KeyExtent that is getting compacted. Is that possible? 

That would be possible and I think its a really good idea.  Doing it in 2.1.1 has the same pros and cons as were discussed with adding a new prop in 2.1.1.  The SPI definition does not prohibit that though.  If there is a compelling enough configurer that could be written with the new method, then maybe that outweighs the possible confusion.

To make the change you suggested I think that could be done by adding key extent [here](https://github.com/apache/accumulo/blob/f3d9903f8072efa50e75970d971595a66dc8affe/core/src/main/java/org/apache/accumulo/core/client/admin/compaction/CompactionConfigurer.java#L50-L56).  Adding to a new method there will not break any code written against 2.1.0, it only breaks things when a developer writes code against 2.1.1 and tries to run it against 2.1.0 where it could cause a method not found exception.